### ISSUE TITLE
WebSocket connection failed: One or more reserved bits

### ISSUE THREAD

**Issue by jwoertink** [cable-cr/cable#7]
When I connect, I'm getting this error in the js console:

```
WebSocket connection to 'ws://localhost:5000/cable?token=JWT_TOKEN' failed: One or more reserved bits are on: reserved1 = 0, reserved2 = 1, reserved3 = 0
```

I'm not using JWT in my app, so I assume the token can just be whatever. I see this in my lucky dev log:

```
web          | GET /cable?token=JWT_TOKEN
web          | I, [2020-11-12 10:35:25 -08:00 #74938]  INFO -- : Started GET "/cable" [WebSocket] for 127.0.0.1:57330 at 2020-11-12 18:35:25 UTC
web          | I, [2020-11-12 10:35:25 -08:00 #74938]  INFO -- : Successfully upgraded to WebSocket (REQUEST_METHOD: GET, HTTP_CONNECTION: Upgrade, HTTP_UPGRADE: websocket)
web          |  ▸ Sent 101 Switching protocols (247.07µs)
web          | I, [2020-11-12 10:35:25 -08:00 #74938]  INFO -- : Finished "/cable" [WebSocket] for 127.0.0.1:57330 at 2020-11-12 18:35:25 UTC
web          |
```

It seems like it's connecting. I don't see any errors on the Crystal side. Redis is running. I'm not sure what else to look for

**Comment by jwoertink** [comment]
a bit more debugging on this. It looks like the socket connects, sends the welcome then disconnects right after. I have a working version in rails and it connects but never disconnects. I'm just not sure why it's disconnecting on the crystal version.

**Comment by jwoertink** [comment]
Found it! https://github.com/cable-cr/cable/blob/master/src/cable/handler.cr#L39 If I remove this line, things start working.... But the question is, why? It seems like the socket actually executes the `on_close` block instead of setting a callback action for what it should do in the case of if it's closes.

**Comment by jwoertink** [comment]
Also had to remove https://github.com/cable-cr/cable/blob/master/src/cable/handler.cr#L46. Once removing that and the `connection.close`, things start working. There's still an issue that's sort of hard to explain, but has to deal with if you have 2 browsers on the channel, and then one refreshes the page, the refreshed page seems to stop getting the socket messages, but the other one still gets them.

**Comment by fernandes** [comment]
Fixed on #3
